os/arch/arm/src/artik053/src/artik053_crashdump.c

```diff
@@ -0,0 +1,178 @@
+
+	for (i = 1; i < strlen(target_str);) {
+		if ((ch = up_getc()) != -1) {
+			host_buf[i] = ch;
+			i++;
+		}
+	}
+
+	if (strncmp(host_buf, target_str, strlen(target_str)) != 0) {
+		/* Send NAK */
+		up_lowputc('N');
+	}
+
+	/* Send ACK */
+	up_lowputc('A');
+
+	/* Send RAM address */
+	ptr = (uint8_t *)&address;
+	for (i = 0; i < sizeof(address); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* Send RAM size */
+	ptr = (uint8_t *)&size;
+	for (i = 0; i < sizeof(size); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* Send RAMDUMP of ram_size bytes */
+	ptr = (uint8_t *)address;
+	while (size) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+		size--;
+	}
+
+	lldbg(" Successfull\n");
+
+	return OK;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_crashdump
+ *
+ * Description:
+ *   If CONFIG_BOARD_CRASHDUMP is selected then up_asseert will call out to
+ *   board_crashdump prior to calling exit in the case of an assertion failure.
+ *   Or in the case of a hardfault looping indefinitely. board_crashdump then
+ *   has a chance to save the state of the machine. The provided
+ *   board_crashdump should save as much information as it can about the cause
+ *   of the fault and then most likely reset the system.
+ *
+ *   N.B. There are limited system resources that can be used by the provided
+ *   board_crashdump function. The tems from the fact that most critical/fatal
+ *   crashes are because of a hard fault or during interrupt processing.
+ *   In these cases, up_assert is running from the context of an interrupt
+ *   handlerand it is impossible to use any device driver in this context.
+ *
+ *   Also consider the following: Who knows what state the system is in? Is
+ *   memory trashed? Is the Heap intact? Therefore all we can expect to do in
+ *   board_crashdump is save the "machine state" in a place where on the next
+ *   reset we can write it to more sophisticated storage in a sane operating
+ *   environment.
+ *
+ ****************************************************************************/
+
+void board_crashdump(uint32_t cur_sp, void *tcb, uint8_t *filename, int lineno)
+{
+	int ret = OK;
+
+#if defined(CONFIG_S5J_WATCHDOG)
+	/* system under panic and we are dumping system state.
+	 * watchdog reset might stall the system, causing crashdump hang.
+	 * Hence let's disable watchdog reset during crashdump process.
+	 */
+	s5j_watchdog_disable();
+#endif
+
+#if defined(CONFIG_BOARD_RAMDUMP_UART)
+	ret = ramdump_via_uart(CONFIG_RAM_START, CONFIG_RAM_SIZE);
+	if (ret != OK) {
+		lldbg("ramdump via uart failed, ret = %d\n", ret);
+	}
+#endif
+
+#if defined(CONFIG_S5J_WATCHDOG)
+	/* we are done !!, let's enable the watchdog reset */
+	s5j_watchdog_enable();
```

Comment:
It depends, if the system is designed to auto reset/reboot even after the device crashed, we might need to put watchdog back to enable state.

Reply:
Then, we had better make a new config for auto reset on crash and use it in here.
As you said, it depens on user choice. But now we reset a board regardless of user opinion.

Reply:
Any way, this is board specific logic, in case of artik053  watchdog was enabled even during assert, So we had disable and enable back.
But if we need to provide a user choice, it should be a seperate  feature/patch

Reply:
Ok. I agree disabling watchdog. But I think we had better stop it in crash status, not reset automatically.
So, my suggetion/opinion is shown below:
1. Let's remove enabling watchdog. It is in crash status so that we don't need to restore original status and it is easier to debug.
2. If we need to support auto reset, then we had better make a new config and support it. As you said, if we need this, we can provide it with sperate patch.

Reply:
Okay, Thanks for your suggestion.
I will remove watchdog enable call.